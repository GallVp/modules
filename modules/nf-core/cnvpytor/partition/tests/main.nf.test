
nextflow_process {

    name "Test Process CNVPYTOR_PARTITION"
    script "../main.nf"
    process "CNVPYTOR_PARTITION"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "cnvpytor"
    tag "cnvpytor/partition"

    test("test-cnvpytor-partition") {

        when {
            process {
                """
                input[0] = [
				    [ id:'test'], // meta map
				    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/pytor/test.pytor', checkIfExists: true)
				]
                input[1] = []

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test-cnvpytor-partition-stub") {
        options '-stub'
        when {
            process {
                """
                input[0] = [
				    [ id:'test'], // meta map
				    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/pytor/test.pytor', checkIfExists: true)
				]
                input[1] = []

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
