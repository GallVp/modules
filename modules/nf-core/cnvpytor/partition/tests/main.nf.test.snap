{
    "test-cnvpytor-partition-stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.pytor:md5,aa03a8fa15b39f77816705a48e10312a"
                    ]
                ],
                "1": [
                    "versions.yml:md5,fbe0931ab8560ef17d80c37c10e9d466"
                ],
                "pytor": [
                    [
                        {
                            "id": "test"
                        },
                        "test.pytor:md5,aa03a8fa15b39f77816705a48e10312a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,fbe0931ab8560ef17d80c37c10e9d466"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T11:10:57.073678"
    },
    "test-cnvpytor-partition": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.pytor:md5,aa03a8fa15b39f77816705a48e10312a"
                    ]
                ],
                "1": [
                    "versions.yml:md5,fbe0931ab8560ef17d80c37c10e9d466"
                ],
                "pytor": [
                    [
                        {
                            "id": "test"
                        },
                        "test.pytor:md5,aa03a8fa15b39f77816705a48e10312a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,fbe0931ab8560ef17d80c37c10e9d466"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T11:09:54.629869"
    }
}